src/services/symbolication.rs

```diff
@@ -1642,70 +1693,55 @@ impl SymbolicationActor {
-                                frame.registers(minidump_state.system_info.cpu_arch),
-                            ),
-                            None => Registers::new(),
-                        };
-
-                        // Trim infinite recursions explicitly because those do not
-                        // correlate to minidump size. Every other kind of bloated
-                        // input data we know is already trimmed/rejected by raw
-                        // byte size alone.
-                        let frame_count = thread.frames().len().min(20000);
-                        let mut frames = Vec::with_capacity(frame_count);
-                        for frame in thread.frames().iter().take(frame_count) {
-                            let return_address =
-                                frame.return_address(minidump_state.system_info.cpu_arch);
-                            module_builder.mark_referenced(return_address);
-
-                            frames.push(RawFrame {
-                                instruction_addr: HexValue(return_address),
-                                package: frame.module().map(CodeModule::code_file),
-                                trust: frame.trust(),
-                                ..RawFrame::default()
-                            });
+                    let timer_old = Instant::now();
+                    let process_state =
+                        ProcessState::from_minidump_breakpad(&minidump, Some(&cfi))?;
+                    let timer_old = timer_old.elapsed();
+                    let (object_infos, stacktraces, minidump_state) = Self::post_process(process_state, cfi_caches_cloned);
+
+                    // Run and time the new stackwalking method if compare_stackwalking_methods was passed
+                    if compare_stackwalking_methods {
+                        let timer_new = Instant::now();
+                        match ProcessState::from_minidump(&minidump, Some(&cfi)) {
+                            Ok(process_state_new) => {
+                                let timer_new = timer_new.elapsed();
+                                let (object_infos_new, stacktraces_new, _) = Self::post_process(process_state_new, cfi_caches);
+
+                                // Report whether the stacktraces are different or identical.
+                                if stacktraces == stacktraces_new && object_infos == object_infos_new {
+                                    metric!(counter("minidump.stackwalk.identical") += 1);
+                                } else {
+                                    metric!(counter("minidump.stackwalk.different") += 1);
+                                }
+                                    metric!(timer("minidump.stackwalk.duration") = timer_old, "method" => "old");
+                                    metric!(timer("minidump.stackwalk.duration") = timer_new, "method" => "new");
+                                }
```

Comment:
i know rustfmt has thrown in the towel, so maybe give it just a little manual help 😄